solutions/0100-0199/0101-symmetric-tree-easy.md

```diff
@@ -62,4 +64,119 @@ public:
         return f(root, root);
     }
 };
-```
\ No newline at end of file
+```
+
+</TabItem>
+
+<TabItem value="java" label="Java">
+<SolutionAuthor name="@vigneshshiv"/>
+
+```java
+/**
+ * Definition for a binary tree node.
+ * public class TreeNode {
+ *     int val;
+ *     TreeNode left;
+ *     TreeNode right;
+ *     TreeNode() {}
+ *     TreeNode(int val) { this.val = val; }
+ *     TreeNode(int val, TreeNode left, TreeNode right) {
+ *         this.val = val;
+ *         this.left = left;
+ *         this.right = right;
+ *     }
+ * }
+ */
+// Time complexity: O(n), where n - # nodes in the tree
+// Space complexity: O(log(n))
```

Comment:
why logn?

Reply:
We can simply say space complexity O(h), h - height of the tree. 

OR

Suppose, we have perfectly balanced binary tree, we start with root that N, then next layer we have left & right subtrees so we have N/2, then next layer is N/4 so on till last. 

Roughly the pattern is

$(N)+(N/2)+(N/4)+(N/8)+... + (N/N)$ which is $O(logN)$ same as height of the tree.

Reply:
I mean state the reason there :D

Reply:
Sure.